File: little-bee/controller/router/signal_threshhold.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_THRESHHOLD = "/config/signal-threshhold"
const URL_ROUTER_SIGNAL_THRESHHOLD_BY_ID = "/config/signal-threshhold/:signal-threshhold-id"
const URL_ROUTER_SIGNAL_THRESHHOLD_QUERY = "/config/signal-threshhold/query"
const URL_ROUTER_SIGNAL_THRESHHOLD_ALL = "/config/signal-threshhold/all"

func SignalThreshholdRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_THRESHHOLD, config.CreateSignalThreshhold)
	r.GET(URL_ROUTER_SIGNAL_THRESHHOLD_BY_ID, config.RetrieveSignalThreshhold)
	r.PUT(URL_ROUTER_SIGNAL_THRESHHOLD, config.UpdateSignalThreshhold)
	r.DELETE(URL_ROUTER_SIGNAL_THRESHHOLD_BY_ID, config.DeleteSignalThreshhold)
	r.POST(URL_ROUTER_SIGNAL_THRESHHOLD_QUERY, config.QuerySignalThreshhold)
	r.GET(URL_ROUTER_SIGNAL_THRESHHOLD_ALL, config.GetAllSignalThreshholds)
	return r
}

File: frontend/console/src/api/signal-threshhold.js
import Axios from "@/http";

export function createSignalThreshhold(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-threshhold', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalThreshhold(signalThreshholdId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-threshhold/${ signalThreshholdId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalThreshhold(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-threshhold', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalThreshhold(signalThreshholdId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-threshhold/${ signalThreshholdId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalThreshholdWithFlag(signalThreshholdId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-threshhold/${ signalThreshholdId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalThreshholds() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-threshhold/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalThreshhold (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-threshhold/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}